perfs_parallel_computing.rst

```diff
@@ -132,7 +189,65 @@ You can have a quick understanding of what happens in the simulation using::
 
   S.ParticleBinning(0).animate()
 
-A ball of plasma is moving through the box.
+A ball of plasma is moving through the box. 
+The box size is described with a 256 x 256 cells grid
+and the plasma ball is a 30 cells radius circle.
+ * With the 8 x 8 patches configuration, the size of a patch is 32 x 32.
+   The plasma, which represents the main time cost, concerns only few patches of the simulation, 16 threads are useless.
+ * With the 16 x 16 patches configuration, the size of a patch is 16 x 16,
+   an order of magnitude is earned regarding the number of patches loaded with particles, almost 1 per thread. 
+ * With the 32 x 32 patches configuration, the size of a patch is 8 x 8,
+   there is more than 3 loaded patches per thread, but with a synchronization overhead.
+
+For this test, in the best case configuration, an additionnal speed-up of 2 is earned.
+This is modest, but accelerating computations needs particles. With a such local plasma, it's hard to achieve.
+
+.. note::
+    That we are investigating new technics based on tasking over species and  clusers of particles to exhibit more shared memory  parallelism.
+
+----
+
+Imbalance 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Without knowing it, you applied load balancing using OpenMP threading.
+Indeed, to achieve good performances using the shared memory parallelism, we apply a ``dynamic`` OpenMP scheduling using the script ``set_omp_env.sh``.
+
+You can observe the difference with the ``static`` scheduling :
+
+.. code-block:: bash
+
+  export OMP_NUM_THREADS=16
+  export OMP_SCHEDULE=static
+  mpirun -np 1 smilei beam_2d.py
+
+With the ``dynamic`` scheduling, threads operate patches one by one in the patches pool while all patches are not treated.
+It's interesting, because the cost of 2 patches regarding particles operation can vary a lot.
+
+With the ``static`` scheduling, the pool is divided with the number of threads, then the first thread operate the first package and so on. 
+If all loaded patches are in the same package, the parallelism is annihilated.
+
+OpenMP offers intermediary solutions but regarding the granularity of the level of parallelism, we advice the ``dynamic`` scheduling.
+
+----
+
+Imbalance and distributed memory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Run the 16 x 16 patches simulation but with a MPI only configuration :
+
+.. code-block:: bash
+
+  source ${SMILEI_ROOT}/scripts/set_omp_env.sh 1
+  mpirun -np 16 smilei beam_2d.py
+
+You can observe that the time spent in the PIC loop is near to the 16 threads ``static`` scheduling time.
+
+.. warning::
+
+   You also may have noticed major differences in sub timers.
+   This is explained by the Smilei's timers implementation which is managed per MPI process,
+   so timers include OpenMP imbalance due to implicit OpenMP barrier of ``#pragma omp for``.
```

Comment:
It is strange to talk about a #pragma here because people will probably have no idea what this is about. Maybe we should mention loops only

Reply:
I completely agree but we have to be informative on these differences.  
I thought that it was the good balance to not fall in too long explanations.  
Again, notes or warnings are open doors for details.